athenzdomains.go

```diff
@@ -0,0 +1,183 @@
+package webhook
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+	"sync"
+
+	"github.com/yahoo/athenz/clients/go/zms"
+	v1 "github.com/yahoo/k8s-athenz-istio-auth/pkg/apis/athenz/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+var (
+	replacer = strings.NewReplacer(".*", ".*", "*", ".*")
+)
+
+// CRMap - cr cache
+type CRMap struct {
+	RoleToPrincipals map[string][]*zms.RoleMember
+	RoleToAssertion  map[string][]SimpleAssertion
+}
+
+// SimpleAssertion - processed policy
+type SimpleAssertion struct {
+	resource *regexp.Regexp
+	action   *regexp.Regexp
+	effect   string
+}
+
+// Cache - cache for athenzdomains CR
+type Cache struct {
```

Comment:
Can we make all structs in this package private along with all their variables? The logic here is all contained within this file / package now we don't need to expose it elsewhere.